cmd/server.go

```diff
@@ -27,16 +31,35 @@ sprayproxy server --backend http://localhost:8081 --backend http://localhost:808
 		viper.AutomaticEnv()
 		host := viper.GetString("host")
 		port := viper.GetInt("port")
+		metricsPort := viper.GetInt("metrics-port")
 		backends := viper.GetStringSlice("backends")
 		insecureSkipTLSVerify := viper.GetBool("insecure-skip-tls-verify")
+		metrics.TLSCRT = viper.GetString("metrics-cert")
+		metrics.TLSKey = viper.GetString("metrics-key")
 		server, err := server.NewServer(host, port, insecureSkipTLSVerify, backends...)
 		if err != nil {
 			return err
 		}
-		return server.Run()
+
+		stopCh := setupSignalHandler()
+		metricsSrvr, err := metrics.BuildServer(metricsPort)
+		if err != nil {
+			return err
+		}
+		go func() {
+			metrics.RunServer(metricsSrvr, stopCh)
+		}()
+		err = server.Run()
+		metrics.StopServer(metricsSrvr)
```

Comment:
(going to create a separate story for this).

Reply:
Filed: https://issues.redhat.com/browse/PLNSRVCE-1081

Reply:
So if I were to use the endless project first noted in that link, I would have to move off of the gin-gonic `Run` call sprayprox makes at https://github.com/redhat-appstudio/sprayproxy/blob/35ce65f1b053028da7477fe80de3f38cf38b3b44/pkg/server/server.go#L50

It is that `Run` method that currently calls `http.ListenAndServe` which is what you replace with `endless.ListenAndServe("host:port", go-gonic-router)`

However, there are useful looking, but private method in `Run` that deal with proxy validation, so that give me pause.

Also, of the 4 alternatives listed there, 3 are from the personal repos of folks.  Always a bit more risky if we ever need a fix.  The last is from facebook, but it is now archived, with no updates since 2018.

For me at least, no slam dunk let's do it now.  

Feels more like a separate story, and not this metrics story, with focused vetting of those approaches, and if need be perhaps forking / copying as permitted into our library, or ROY, based on where we land around support for those alternatives.

And if we end up agreeing on ^^, what to do with what is here now:
1) leave the signal handling for metrics at least for now
2) or back that out and leave ungraceful for both